### PR TITLE
Created auth interceptor for use with Angular's HttpClient

### PR DESCRIPTION
**Pull request by Blackbaud-PaulCrowder**
The implementation contains a workaround for passing additional context to the interceptor since Angualr doesn't currently support it.  Here is an issue describing the problem: https://github.com/angular/angular/issues/18155

### CODE REVIEW

File: src/app/public/modules/auth-http/auth-interceptor.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+import {
+  SKY_AUTH_PARAM_AUTH,
+  SKY_AUTH_PARAM_PERMISSION_SCOPE
+} from './auth-interceptor-params';
+
+//#endregion
+
+function removeSkyParams(request: HttpRequest<any>): HttpRequest<any> {
+  /* istanbul ignore else */
```
Comment (Blackbaud-BobbyEarl) [inline src/app/public/modules/auth-http/auth-interceptor.ts]: Any reason to chose this method over having a test that doesn't supply the `param` property?

Comment (Blackbaud-PaulCrowder) [reply src/app/public/modules/auth-http/auth-interceptor.ts]: I verified that `params` is always defined, even when not specified on the original HTTP request.  I left the `if` statement and ignore comment in and added a comment above it to indicate why it's being ignored.

**Review by Blackbaud-SteveBrush — APPROVED** [review]
Tested. Works great; code looks great.

**Comment by Blackbaud-SteveBrush** [comment]
@blackbaud-johnly We'll need a documentation task for this, please.

**Comment by blackbaud-johnly** [comment]
Created https://github.com/blackbaud/skyux2/issues/2187 to document this.

**Pull request merged**